### ISSUE TITLE
Article Editor Can Allows User to Change Permissions and Topics which would prevent them from previewing or Saving

### ISSUE THREAD

**Issue by eSilverStrike** [Geeklog-Core/geeklog#1135]
Article Editor does allow users to change the article permissions and Topics which would prevent them from previewing or saving the article.

For example if the article has Group Edit enabled and it is removed or the owner of the article removes their edit access when they try to then preview or save the article they get kicked out of the editor due to lack of permissions.

Permissions for the article is loaded from the editor not from the database.

`$result = $story->loadFromArgsArray($_POST);`

I think it should load permissions (if existing article) from the database as technically they still have permissions until after the article is saved with any changes.

Topics appear to work the correct way and will allow save or preview if the user selects a topic they do not have edit access to. Once saved they then cannot reedit the article (as one of the topics it is assigned to they do not have access). The article list may have an issue here though (further testing needed).

**Change Default Permissions?**
Default Article and Topic Permissions give the owner read and edit access.

Default Article and Topic Permissions give the Group only read access. I think by default they should get read and edit access since they are the Story and Topic Admin group? It's much more likely that the Story Admin should have edit access to all articles and not just the owners. Thoughts?

**Comment by eSilverStrike** [comment]
@mystralkk This issue has been in Geeklog for a long time. I don't think we should get into this issue now before the 2.2.2 release  unless you see an easy fix? I added this to the 2.2.3 milestone. Thoughts?

**Comment by mystralkk** [comment]
No, I don't see any easy fix for this.  Since this issue has been around for a long time, a bit delay will not make much difference.

**Comment by eSilverStrike** [comment]
Agree. I would even consider pushing it even later as other items like Topics or Staticpages could suffer the same issue on save (they do not have preview so that would not be an issue)
